src/components/dace_interface.ts

```diff
@@ -680,39 +756,20 @@ export class DaCeInterface {
     }
 
     @ICPCRequest()
-    public writeToActiveDocument(json: JsonSDFG | string): void {
-        const activeEditor = DaCeVSCode.getInstance().getActiveWebview();
+    public async writeToActiveDocument(json: JsonSDFG | string): Promise<void> {
+        const activeEditor = DaCeVSCode.getInstance().activeSDFGEditor;
         if (activeEditor) {
-            const sdfvInstance = SdfgViewerProvider.getInstance();
-            const document = sdfvInstance?.findEditorForWebview(
-                activeEditor
-            )?.document;
-            if (document) {
-                const edit = new vscode.WorkspaceEdit();
-                if (typeof json === 'string')
-                    edit.replace(
-                        document.uri,
-                        new vscode.Range(0, 0, document.lineCount, 0),
-                        JSON.stringify(JSON.parse(json), null, 2)
-                    );
-                else
-                    edit.replace(
-                        document.uri,
-                        new vscode.Range(0, 0, document.lineCount, 0),
-                        JSON.stringify(json, null, 2)
-                    );
-                vscode.workspace.applyEdit(edit);
-            }
+            // TODO: respect indent size for JSON serialization?.
```

Comment:
no. It's hardcoded in dace and I'd even use 1 to make the file size smaller

Reply:
Adapted to that too